Assets/_BForBoss/_VFX/Scripts/TimedVFXEffect.cs

```diff
@@ -32,7 +34,10 @@ public void StartEffect()
         private async void StopAfterTime()
```

Comment:
Just realized, I'm not sure why this needs to be async? (instead of coroutine). If this was just happening once, then it wouldn't make a difference, but `async` is quite heavy on garbage collection in comparison to coroutines ([reference](https://www.linkedin.com/pulse/unity-async-vs-coroutine-jo%C3%A3o-borks/#:~:text=While%20analyzing%20allocation,of%20operations%20simultaneously.))

And since this is being used on every enemy, it might be worth to switch?

Reply:
Updated to be a coroutine